src/ui.rs

```diff
@@ -116,6 +167,20 @@ impl Ui {
         }
     }
 
+    /// Switches the output to use the pager, if allowed.
+    pub fn request_pager(&mut self) {
+        if !matches!(self.paginate, PaginationChoice::Auto) {
```

Comment:
You should be able to use `!=` here instead of `!matches!()`.

Also, I think returning if `self.paginate == PaginationChoice::Never` would be more obvious. When I saw this code, I was confused why the `Always` case wasn't handled here.

Reply:
Separate from the fact that `Pagination::Always` _should_ imply that the pager is already active, I vaguely remember my mental model being that `request_pager` only makes sense with `PaginationChoice::Auto`, but what you said makes more sense to me right now.

Reply:
This confused me a little at first but I worked out the intention you described quickly. We probably don't want to spawn a new pager if one already exists. Maybe just comment?

Reply:
We won't spawn a pager anyway if `self.output` is a `UiOutput::Paged`. I prefer comparing to `Never` because it's more future-proof, but an assertion would also work (safer than a comment)

Reply:
Yeah I think comparing to `Never` is the most obvious approach.